android/src/com/mapswithme/maps/maplayer/MapButtonsController.java

```diff
@@ -0,0 +1,234 @@
+package com.mapswithme.maps.maplayer;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import com.mapswithme.maps.R;
+import com.mapswithme.maps.routing.RoutingController;
+import com.mapswithme.maps.routing.SearchWheel;
+import com.mapswithme.maps.widget.menu.MyPositionButton;
+import com.mapswithme.maps.widget.placepage.PlacePageController;
+import com.mapswithme.util.Config;
+import com.mapswithme.util.Graphics;
+import com.mapswithme.util.UiUtils;
+
+public class MapButtonsController
+{
+  @NonNull
+  private final View mButtonsFrame;
+  @NonNull
+  private final View mZoomFrame;
+  @NonNull
+  private final ImageButton mLayersButton;
+  @NonNull
+  private final View myPosition;
+  @NonNull
+  private final View mSearchButtonFrame;
+  @Nullable
+  private final MyPositionButton mNavMyPosition;
+  @NonNull
+  private final MapLayersController mToggleMapLayerController;
+  @NonNull
+  private final SearchWheel mSearchWheel;
+  private final PlacePageController mPlacePageController;
+  private final float mBottomMargin;
+  private final int mButtonWidth;
+  private final float mInitialButtonMargin;
+  private float mTopLimit;
+  private float mContentHeight;
+  private float mContentWidth;
+
+  public MapButtonsController(@NonNull View frame, AppCompatActivity activity, MapButtonClickListener mapButtonClickListener, PlacePageController placePageController)
+  {
+    mButtonsFrame = frame.findViewById(R.id.navigation_buttons_inner);
+    mZoomFrame = frame.findViewById(R.id.zoom_buttons_container);
+    mPlacePageController = placePageController;
+    View zoomInButton = frame.findViewById(R.id.nav_zoom_in);
+    zoomInButton.setOnClickListener((v) -> mapButtonClickListener.onClick(MapButtons.zoomIn));
+    View zoomOutButton = frame.findViewById(R.id.nav_zoom_out);
+    zoomOutButton.setOnClickListener((v) -> mapButtonClickListener.onClick(MapButtons.zoomOut));
+    myPosition = frame.findViewById(R.id.my_position);
+    mNavMyPosition = new MyPositionButton(myPosition, (v) -> mapButtonClickListener.onClick(MapButtons.myPosition));
```

Comment:
(v)?

Reply:
`OnClickListener` expects a function accepting a `View` and not returning anything. If I do not put the view argument the compiler complains. So here the `v` argument is simply ignored. I did not find a better way, seems like a limitation from Java.